apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PageLinkBuilder.java

```diff
@@ -179,41 +155,27 @@ public void addPageLinks(Bundle toBundle, int numTotalResults) {
    */
   private String createPageLink(int startIndex) {
     StringBuilder b = new StringBuilder();
+
+    // Setup URL base and resource.
     b.append(serverBase);
     b.append(resource);
-    b.append(Constants.PARAM_COUNT + "=" + getPageSize());
-    b.append("&startIndex=" + startIndex);
-    b.append("&" + searchByDesc + "=" + identifier);
-
-    // Add the lastUpdated parameters if present
-    if (lastUpdated != null) {
-      DateParam lowerBound = lastUpdated.getLowerBound();
-      if (lowerBound != null && !lowerBound.isEmpty()) {
-        b.append(
-            "&"
-                + Constants.PARAM_LASTUPDATED
-                + "="
-                + lowerBound.getPrefix().getValue()
-                + lowerBound.getValueAsString());
-      }
-      DateParam upperBound = lastUpdated.getUpperBound();
-      if (upperBound != null && !upperBound.isEmpty()) {
-        b.append(
-            "&"
-                + Constants.PARAM_LASTUPDATED
-                + "="
-                + upperBound.getPrefix().getValue()
-                + upperBound.getValueAsString());
-      }
-    }
 
-    // Add the "type" parameter if present
-    if (claimTypes != null) {
-      b.append("&type=" + claimTypes.toString().toLowerCase().replaceAll("[\\[\\] ]", ""));
-    }
+    // Get a copy of all request parameters.
+    Map<String, String[]> params = new HashMap<String, String[]>(requestDetails.getParameters());
```

Comment:
Small suggestion. You can make this read simpler by removing...
`Map<String, String> param = new HashMap<>(requestDetails...`